open_fdd/gateway/server.py

```
@@ -1000,44 +1131,285 @@ def onboard_config_set(body: OnboardConfigBody) -> dict[str, Any]:
+            cleaned = ["csv"]
+        try:
+            merged, used_sources = services.ingest.load_merged_sources_frame_window(
+                site_id=body.site_id,
+                sources=cleaned,
+                start_ts=body.start_ts,
+                end_ts=body.end_ts,
+                join_how=str(body.join_how),
+            )
+        except ValueError as exc:
+            raise HTTPException(status_code=400, detail=str(exc)) from exc
+        safe_rules = _safe_rules_path(body.rules_path)
```

Ruff (0.15.12)

[error] Undefined name `_safe_rules_path`

(F821)